lib/BellCheck.ml

```diff
@@ -245,18 +245,21 @@ module Make
             code)
         parsed.MiscParser.prog in
       let parsed = { parsed with MiscParser.prog; } in
-      let test_bi = parsed.MiscParser.extra_data in
-      let test_bi = match test_bi with
-      | MiscParser.BellExtra b -> b
-      | _ -> Warn.fatal "Error getting bell information from test" in
+      let test_bi =
+        let test_bi = parsed.MiscParser.extra_data in
+        List.hd (List.map
+          (function
+           | MiscParser.BellExtra b -> b
+           | _ -> Warn.fatal "Error getting bell information from test")
+          test_bi) in
```

Comment:
Using `List.hd` is dangerous. If for some reason (_e.g._ a bug) the list is empty, users are faced with anon specific exceptions. Here is  a more robust alternative:
```suggestion
        let test_bi = parsed.MiscParser.extra_data in
        match (List.map
          (function
           | MiscParser.BellExtra b -> b
           | _ -> Warn.fatal "Error getting bell information from test")
          test_bi) with
          | [] -> assert false
          | x::_ -> x in
```
That amounts to inlining `List.hd`.